src/DebuggerData.cpp

```diff
@@ -180,8 +180,7 @@ void Breakpoints::setBreakpoints(const QString& str)
 		int q = bp.lastIndexOf('{');
 		if (bp.mid(q).simplified() != "{debug break}") continue;
 
-		newBp.condition = bp.mid(p, q - p).simplified();
-		unescapeXML(newBp.condition);
+		newBp.condition = unescapeXML(bp.mid(p, q - p).simplified());
```

Comment:
I know you only transformed the existing code. But I wonder if using `simplified()` is correct here.

Suppose the condition is something like:
```$::myString eq "two  spaces"```
Then `simplified()` will reduce this to only have a single space between `two` and `spaces`.

I think it would be safer to replace `simplified()` with `trimmed()`. In other words: keep the condition-string (a Tcl expression) fully unchanged.

Reply:
This has potentially larger implications in the code. For instance, one could add a breakpoint with a rule such as `myGlobalString eq "two⠀⠀spaces"` and `createSetCommand()` will trim it down inappropriately because it also uses `simplified()` internally.

Reply:
We may need to do a global audit of all uses of `simplified()`.
I don't expect you to do this work as part of this PR. But of course you're very welcome if you do want to tackle it.